bundler/spec/other/major_deprecation_spec.rb

```diff
@@ -104,9 +104,9 @@
     pending "is removed and shows a helpful error message about it", bundler: "3"
   end
 
-  describe "bundle update --quiet" do
+  describe "bundle update --all --quiet" do
     it "does not print any deprecations" do
-      bundle :update, quiet: true, raise_on_error: false
+      bundle :update, all: true, quiet: true, raise_on_error: false
```

Comment:
In Bundler 3, _not_ passing `--all` (or an explicit list of gems) is considered deprecated. This spec was erroneously passing because it was exiting (without a deprecation warning) with an error that the Gemfile had no specs _before_ the deprecation check. Due to re-arranging in that file to defer loading the Definition until after the deprecation check (as a simple matter of sharing arguments between the plugin run and the real run of evaluating the gemfile), the deprecation warning was getting tripped, and thus failing this spec.

Reply:
That makes total sense, thanks for explaining!